accepted/2024/[TS]PMUI-Readme-rendering.md

```diff
@@ -0,0 +1,116 @@
+
+If the README resource has been added to the service index then we will use the URL format defined to download the README.
+This will involve checking that URL for every package. 
+
+If neither of these options are available we will use look for the README in the local file system.
+Meaning we will only render the README for already downloaded packages.
+
+It will be up to the individual feeds to implement a solution that works best for them, or even implement both. 
+
+## Drawbacks
+MarkdownPreview control currently marked as obsolete since the interface has not been finalized.
+So when an upgrade is made we may have to change how we use the control.
+
+WebView2 controls always render ontop of other controls in the view.
+[Secnario 25254665](https://microsoft.visualstudio.com/Edge/_workitems/edit/25254665).
+PM UI needs to be updated to ensure items don't scroll off screen.
+
+## Rationale and alternatives
+By using an existing control we maintain consistency throughout the IDE and can rely on the owner to fix any bugs with the control.
+
+Due to concerns about performance we will not be downloading the full nupkg temporarily just to access the README.
+
+## Prior Art
+The IMarkdownPreview is currently being used when creating a new pull request inside of Visual Studio.
+![Alt text](../../meta/resources/READMEPMUI/PullRequestExperience.png) 
+
+nuget.org currently renders the README and our users will expect them to look the same.
+Ex. https://www.nuget.org/packages/Newtonsoft.Json#README-body-tab
+
+## Unresolved Questions
+1. Will we show README in the updates tab?
+1. ~~What do we show if the package has a README.txt instead of md?~~
+    * The README is spec is written as only accepting MD.
+    So we will use the nuspec to determine where the README is and treat it as md, even if the file is actually txt. 
+1. ~~What do we show if there is no README defined?~~
+    * Show the README tab with a message saying there is no README for the selected pacakge/version.
+1. ~~Where do we get the README from when it's not on the disk?~~
+    - Will document two ways for servers to provide a the readme.
+    Updating the package metadata to include a link to the readme, and a direct download similar to the way icon is implemented.
+1. ~~Where are the README files saved in a package?~~
+    - Can use Nuget.Packaging to get README location from nuspec.
+1. What do we want the UX to be when an exception or error occurs while reading a README file? 
+1. ~~Do we want the README to update whenever a new version is selected for the current package?~~
+   - Yes.
+<!-- What parts of the proposal do you expect to resolve before this gets accepted? -->
+<!-- What parts of the proposal need to be resolved before the proposal is stabilized? -->
+<!-- What related issues would you consider out of scope for this proposal but can be addressed in the future? -->
+
```

Comment:
Is that tab selection of readme vs details going to be sticky?

Reply:
Yup, right now it keeps the state between tabs and opening/closing the PM UI